### PR TITLE
CAMEL-11335: Add Actuator endpoint for exposing Camel routes

### PR DESCRIPTION
**Pull request by gzurowski**
See https://issues.apache.org/jira/browse/CAMEL-11335

### CODE REVIEW

**Comment by zregvart** [comment]
Awesome work 👍, I would only consider extending from [AbstractEndpoint](http://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/actuate/endpoint/AbstractEndpoint.html) for the implementation of `isEnabled` that takes environment into account.

**Comment by davsclaus** [comment]
You may also want to display the state of the route, whether its started / stopped etc

**Comment by gzurowski** [comment]
Thanks for your feedback @zregvart @davsclaus.

Good point about the [AbstractEndpoint](http://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/actuate/endpoint/AbstractEndpoint.html), I will update this in a bit.

I have also planned to get more route details into the Actuator endpoint, just wanted to get your general feedback first in order to get this rolling, and extend with more details later.

File: components/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/actuate/endpoint/CamelRoutesEndpoint.java [hunk 1]
```diff
@@ -25,37 +25,24 @@
...
-public class CamelRoutesEndpoint implements Endpoint<List<RouteEndpointInfo>> {
+public class CamelRoutesEndpoint extends AbstractEndpoint<List<RouteEndpointInfo>> {
 
-    private static final String ENDPONT_ID = "camelroutes";
+    private static final String ENDPOINT_ID = "camelroutes";
 
     private CamelContext camelContext;
 
     public CamelRoutesEndpoint(CamelContext camelContext) {
+        super(ENDPOINT_ID);
```
Comment (zregvart) [inline components/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/actuate/endpoint/CamelRoutesEndpoint.java]: Quick note: I think you want to use `super(ENDPOINT_ID, true)` to make the Endpoint sensitive (as was before).

Comment (gzurowski) [inline components/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/actuate/endpoint/CamelRoutesEndpoint.java]: Thanks for pointing it out, but already fixed in the latest updates where I extend from AbstractEndpoint.

**Comment by gzurowski** [comment]
I have extended from [AbstractEndpoint](http://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/actuate/endpoint/AbstractEndpoint.html) as suggested, and would go forward and merge into master, if that is OK with you. I will follow up with commits that included more route details with another PR.

**Comment by oscerd** [comment]
+1 for me. Great work.

**Comment by davsclaus** [comment]
@gzurowski a good idea after merging to master branch is to rebase on top so the git tree is a straight line and your code changes is applied on latest code.

eg run

    git pull --rebase

**Pull request merged**